cmd/webhook/main.go

```diff
@@ -167,7 +242,9 @@ func main() {
 	sharedmain.WebhookMainWithContext(ctx, "webhook",
 		certificates.NewController,
 		newDefaultingAdmissionController,
```

Comment:
You can just add domain mapping types to the `newDefaultingAdmissionController` and `newValidationAdmissionController` and you don't need to create new ones

Reply:
@dprotaso I thought about that. My idea was to keep a separate controller per concern for now, to be able to have some independent configuration in the feature (eg. worker per queue), logging etc.

Reply:
> If we merge then should not we increase the concurrency for the common controller (default is 2)?

There really isn't an advantange here - this admission controller is reconciling a single resource

Reply:
Yes but is not the case that the number of reconciled objects can be equal to the number of services, since a domain mapping can be set per namespace? So the load for the domain reconciler is not low no?

Reply:
I'm confused - this controller just mutates this one resource here https://github.com/knative/serving/blob/main/config/core/webhooks/defaulting.yaml

> So the load for the domain reconciler is not low as far as I can tell eg. during a restart. Should we merge the controllers and keep the concurrency defaults? Anyway I will do a quick test.

Increasing the # of controllers will bump the total QPS & Burst (it's code in the sharedmain). As per reconciler there are two go routines - so this doesn't change when moving a controller/reconciler from one go cmd to another.

Reply:
I am talking about the reconciliation work done and the workqueue workers, not the rate with which we hit the api server (the latter is fine).

Reply:
> So if we move the domain mapping types for defaulting to the default controller then we will still have 2 routines but with more types to handle in case of reconciliation, no?

More types in the webhook != more resources that need reconciliation. We should only have one defaulting webhook and one validation webhook.

It's this one here
https://github.com/knative/serving/blob/921daf867a3c598246836bcd2f9451ccdc10b0d0/config/core/webhooks/defaulting.yaml#L34-L53

Actually we should be moving these values from this webhook over to the new one and just exclude the `/status`
https://github.com/knative/serving/blob/921daf867a3c598246836bcd2f9451ccdc10b0d0/config/core/webhooks/domainmapping-defaulting.yaml#L42-L44


> If we merge these controllers then we have 2 routines by default but with potentially more keys enqueued and more work for the common work queue, no?

Generally we haven't cared about this but it's not an issue for the webhook controllers since it's only reconciling a single resource. There are better gains to make elsewhere - for example the revision controller still has a concurrency of two - maybe the serverlessservice (since it deals with endpoint updates) could be bumped higher.

But I'd say that's out of scope for this PR.

Reply:
I guess to clarify I'm referring to the 'threadiness' per controller out of scope for this PR. But I still think it makes sense to consolidate domain mapping into the default webhook